packages/api/cms-api/src/dam/files/folders.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Controller, Get, Param, Res } from "@nestjs/common";
+import { Response } from "express";
+
+import { FoldersService } from "./folders.service";
+
+@Controller("dam/folders")
+export class FoldersController {
+    constructor(private readonly foldersService: FoldersService) {}
+
+    @Get("/:folderId/zip")
+    async createZip(@Param("folderId") folderId: string, @Res() res: Response): Promise<void> {
+        const folder = await this.foldersService.findOneById(folderId);
```

Comment:
This should fail if the folder can't be found. You'd then won't need the fallback values for the folder name